models/protocols/SendGridProtocol.cfc

```
@@ -127,25 +127,31 @@ component extends="cbmailservices.models.AbstractProtocol" {
                 } );
         }
 
-        cfhttp( url = "https://api.sendgrid.com/v3/mail/send", method = "POST" ) {
-            cfhttpparam( type = "header", name = "Authorization", value="Bearer #getProperty( "apiKey" )#" );
-            cfhttpparam( type = "header", name = "Content-Type", value="application/json" );
-            cfhttpparam( type = "body", value = serializeJson( body ) );
-        };
+        try {
+            cfhttp( url = "https://api.sendgrid.com/v3/mail/send", method = "POST", result = "sendGridResponse", timeout = 10 ) {
```

I believe that without `throwOnError="true"` this will never throw.

---

Just added it

---

Sorry, to be clear, I don't think you want `throwOnError`.  That turns 4xx and 5xx codes into exceptions which would skip your handling below.  I was more saying you don't need the `try`/`catch`.

---

What the try/catch does is capture/log any issue that does not return a response, which would not be caught or handled by the logic below the cfhttp call.   The issue I am troubleshooting right now is a silent failure - no errors, no debug information.  So, I am trying to make sure any possible issue is caught and logged.